### PR TITLE
🐛 Fixed command parameters types

### PR DESCRIPTION
**Pull request by Kylianalex**
was not working and everything was str
also bool("False") will return True because the string is not empty

### Changes

-   `improvements`: now a parameter with type int will return an int, not str

 ### Check off the following

-   [x] I have tested my changes with the current requirements
-   [x] My Code follows the pep8 code style.

### CODE REVIEW

File: pincer/objects/app/interactions.py [hunk 1]
```diff
@@ -145,33 +145,36 @@ def __post_init__(self):
...
+            if option.type == AppCommandOptionType.STRING:
                 option.value = str(option.value)
-            elif option.type is AppCommandOptionType.INTEGER:
+            elif option.type == AppCommandOptionType.INTEGER:
                 option.value = int(option.value)
-            elif option.type is AppCommandOptionType.BOOLEAN:
-                option.value = bool(option.value)
-            elif option.type is AppCommandOptionType.NUMBER:
+            elif option.type == AppCommandOptionType.BOOLEAN:
+                option.value = option.value == "True"
```
Comment (Sigmanificient) [inline pincer/objects/app/interactions.py:153]: I'm not sure of this change

File: pincer/objects/app/interactions.py [hunk 2]
```diff
@@ -145,33 +145,36 @@ def __post_init__(self):
             return
 
         for option in self.data.options:
-            if option.type is AppCommandOptionType.STRING:
+            if option.type == AppCommandOptionType.STRING:
```
Comment (Sigmanificient) [inline pincer/objects/app/interactions.py:148]: @Lunarmagpie not sure, but this is alright?

File: pincer/objects/app/interactions.py (hunk 1 above)
Comment (Kylianalex) [reply pincer/objects/app/interactions.py:153]: the problem is that discord send "True" or "False" as str and i don't know how to make it boolean with the pep8 style

File: pincer/objects/app/interactions.py (hunk 2 above)
Comment (Kylianalex) [reply pincer/objects/app/interactions.py:148]: #425 the "is" statement is not working for me in python 3.9

Comment (Arthurdw) [reply pincer/objects/app/interactions.py:148]: Okay, I'm assuming that our APIObject isn't properly converting the socket payload to an object. As I'm not up to date with the current state of the way the lib handles this, I can't provide a solution rn.  
  
One thing I do know is that this is not the expected behavior, and we should model around the expected behavior _(which is that there is an object, so a `is`_ statement should be used_). Nonetheless,_ this issue should be addressed.

**Pull request closed without merging**